src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -44,46 +43,127 @@ private class ConfigurationCacheItem
         public FeatureManager(
             IFeatureDefinitionProvider featureDefinitionProvider,
-            IEnumerable<IFeatureFilterMetadata> featureFilters,
-            IEnumerable<ISessionManager> sessionManagers,
-            ILoggerFactory loggerFactory,
-            IOptions<FeatureManagementOptions> options,
-            IOptions<TargetingEvaluationOptions> assignerOptions)
+            FeatureManagementOptions options)
         {
-            _featureDefinitionProvider = featureDefinitionProvider;
-            _featureFilters = featureFilters ?? throw new ArgumentNullException(nameof(featureFilters));
-            _sessionManagers = sessionManagers ?? throw new ArgumentNullException(nameof(sessionManagers));
-            _logger = loggerFactory.CreateLogger<FeatureManager>();
-            _assignerOptions = assignerOptions?.Value ?? throw new ArgumentNullException(nameof(assignerOptions));
+            _featureDefinitionProvider = featureDefinitionProvider ?? throw new ArgumentNullException(nameof(featureDefinitionProvider));
+            _options = options ?? throw new ArgumentNullException(nameof(options));
             _filterMetadataCache = new ConcurrentDictionary<string, IFeatureFilterMetadata>();
             _contextualFeatureFilterCache = new ConcurrentDictionary<string, ContextualFeatureFilterEvaluator>();
-            _options = options?.Value ?? throw new ArgumentNullException(nameof(options));
-            _parametersCache = new MemoryCache(new MemoryCacheOptions());
+            _featureFilters = Enumerable.Empty<IFeatureFilterMetadata>();
+            _sessionManagers = Enumerable.Empty<ISessionManager>();
+            _assignerOptions = new TargetingEvaluationOptions();
+        }
+
+        /// <summary>
+        /// The collection of feature filter metadata
+        /// </summary>
+        public IEnumerable<IFeatureFilterMetadata> FeatureFilters
+        {
+            get => _featureFilters;
+
+            init
+            {
+                _featureFilters = value ?? throw new ArgumentNullException(nameof(value));
+            }
+        }
+
+        /// <summary>
+        /// The collection of session managers
+        /// </summary>
+        public IEnumerable<ISessionManager> SessionManagers
+        {
+            get => _sessionManagers;
+
+            init
+            {
+                _sessionManagers = value ?? throw new ArgumentNullException(nameof(value));
+            }
         }
 
+        /// <summary>
+        /// Application memory cache to store feature filter settings
+        /// </summary>
+        public IMemoryCache Cache { get; init; }
+
+        /// <summary>
+        /// The logger for the feature manager
+        /// </summary>
+        public ILogger Logger { get; init; }
+
+        /// <summary>
+        /// The collection of telemetry publishers
+        /// </summary>
         public IEnumerable<ITelemetryPublisher> TelemetryPublishers { get; init; }
```

Comment:
How come this one is nullable?

Reply:
https://github.com/microsoft/FeatureManagement-Dotnet/blob/preview/src/Microsoft.FeatureManagement/FeatureManager.cs#L674

We will log warning if the feature flag asks to publish telemetry but there is no registered telemetry publisher.
``` C#
if (TelemetryPublishers == null || !TelemetryPublishers.Any())
{
    _logger.LogWarning("The feature declaration enabled telemetry but no telemetry publisher was registered.");
 }
```

Reply:
We can simplify to just `!TelemetryPublishers.Any` like our other `IEnumerable` usage.